frontend/src/components/BattleLog.tsx

```diff
@@ -20,10 +21,22 @@ const BattleLog: React.FunctionComponent<Props> = ({
   enemyUnits,
   damageUnit,
 }: Props) => {
+  const allUnits = [...units, ...enemyUnits];
+  const whos = log.map(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    action => allUnits.find(unit => unit.id === action.who)!,
+  );
+  const whoms = log.map(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    action => allUnits.find(unit => unit.id === action.whom)!,
+  );
+
   const [currentLog, setCurrentLog] = useState<Array<BattleAction>>([]);
   const lastElement = useRef<HTMLDivElement>(null);
   const [lastExecute, setLastExecute] = useState<number>(performance.now());
 
+  const projectileDuration = log.length / actionDuration;
```

Comment:
Hmm... Why don't you just make it actionDuration? Imagine action duration is 200ms, and log length is 1000. Then it's 1000 / 200 = 5s. Which makes the last few bullets go somewhere into space.

Reply:
Oh, this is very obviously my bad... I thought actionDuration was the duration of the round and I accidentally put the division the wrong way xd